peercrawler.py

```diff
@@ -82,6 +82,31 @@ def count_good_peers(self):
     def count_peers(self):
         return len(self.get_peers_copy())
 
+    def listen_incoming(self):
+        with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as s:
+            s.bind(("::", 7075))
+            s.listen()
+
+            while True:
+                connection, address = s.accept()
+                threading.Thread(target=self.handle_incoming, args=(connection, address), daemon=True).start()
```

Comment:
There needs to be a limit to the number of threads that can be spawned, it cannot be limitless.
Create a counting semaphore, take the for every thread created and release it when the thread does.